proto/note.proto

```diff
@@ -23,6 +23,11 @@ message Note {
     optional bytes details = 6;
 }
 
+message NoteInclusionProof {
+    digest.Digest note_id = 1;
+    merkle.MerklePath merkle_path = 2;
+}
```

Comment:
This needs to include a bit more data. Specifically, note inclusion proof should be sufficient to prove that a note is present in the chan. This requires:

1. A Merkle path from the leaf to the root of note Merkle tree of a block.
2. Index of the note in this block.
3. Block's note root + sub hash (or alternatively, the entire block header).
4. MMR path to the block in the current chain MMR.

Since several notes could be from the same block, it may make sense to de-duplicate the data. Though, maybe it doesn't matter too much and at least for now we can send all required data on per note basis.

Reply:
@bobbinth thank you for clarification!
> 4. MMR path to the block in the current chain MMR.
Do we need this field?
I found implementation of `NoteInclusionProof` in `miden-base` and there is no such field:
https://github.com/0xPolygonMiden/miden-base/blob/f80276e93521f896b2ff5544f42ddaba6e50038e/objects/src/notes/origin.rs#L23-L35